resources/templates/md/posts/2016-04-04-state.md

```diff
@@ -0,0 +1,366 @@
+{:title "The approach to state management in Clojure"
+:description  "Intro to mutable state in Clojure. Greg the llama show you an effective way how to manage state, and you'll find out that llamas love candies too :)"
+:layout :post
+:tags  ["Clojure" "ClojureScript" "Fundamentals"]
+:toc false
+:author "Dajana Štiberová"
+:image ""}
+
+Long time ago, during one interview, I was asked to describe the difference between `vars` and `atoms`.
+It was a pretty simple question, but at that time it was not so easy to describe this difference for me.
+Now I'm more skilled in Clojure but I still think that it was an interesting question and when
+I was looking for a topic for the blog post, I decided to use this one.
+
+So if you want to read something about state management, or you want to know how to dance like Greg the llama, this post is for you :)
+
+## States and Identities
+
+We need to know that in Clojure there is a difference between state and identity.
+
+`States` are an immutable data.
+A state is the value an identity points to at a particular time. State never changes.
+
+`Identities` unify different values over the time.
+Identity types are mutable references to immutable values,
+and can be modelled by using one of the four references types (`Var`s, `Atom`s, `Agent`s, `Ref`s)
+
+
+### *Example*
+
+Our example is the best llama on earth, Greg the llama.
+Greg is an identity.
+Greg wants to go to the party with Gregorina. He is a good dancer, and he wants to teach us how to dance.
+
+So, Greg is dancing and every snapshot of this action is one state.
+
+Greg has a series of different values (moves) over time (states 1-4).
+We need to realize, that we can transition the state but it is not possible to change or modify the previous state.
+The identity (Greg) can be modelled by using some reference type pointing to the value,
+and subsequently calling appropriate functions to create new states.
+
+NOTE:
+
+If you don't have any knowledge about threads in Clojure, you should read about [threads](http://blog.clojuregirl.com/clojure_threads.html) first.
+
+## How we can manage states (or how the data might change over time)
+
+There are four ways how to model an identity (manage states).
+Using reference types: `Vars` , `Atoms` , `Refs` , `Agents` .
+
+Greg has a girlfriend Gregorina and they go to the party. It's halloween party and
+there are a lot of sweets everywhere. Llamas love candies and as we know, 
+favourite candies for llamas are yummy marshmallows and gummy bears.
+
+![marshmallow](../img/marshmallow.png "Marshmallow")
```

Comment:
Overall layout: Here I would align the text _next to_ the image. Now there is too much empty space. It is an easy CSS change.
<img width="516" alt="screen shot 2016-04-12 at 19 38 14" src="https://cloud.githubusercontent.com/assets/15321488/14469719/58b7f116-00e6-11e6-9aad-c76f1ac1de8a.png">